src/hera/workflows/_meta_mixins.py

```diff
@@ -372,6 +372,12 @@ def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
                 return Step(template=self, **kwargs)
 
             if isinstance(_context.pieces[-1], DAG):
+                # Add dependencies based on context if not explicitly provided
```

Comment:
* Note this is different from the implementation in the experimental decorator code, which does not allow a `depends` to be passed at all: https://github.com/argoproj-labs/hera/blob/2eec4c282557823c661dae6b410b470d13a90be5/src/hera/workflows/_meta_mixins.py#L568-L569
* I think this logic should probably be made common -- perhaps moved into DAG?

Reply:
> Would it be reasonable for the existing code to change to match?

Yes that would be good! 🙏

Reply:
Done.